content/storage-history/history.textile

```diff
@@ -416,6 +456,35 @@ channel.on(.attached) { error in
 }
 ```
 
+```[realtime_go]
+client, err := ably.NewRealtime(
+  ably.WithKey("{{API_KEY}}"))
+if err != nil {
+  log.Fatalf("Failed to create Ably Realtime client: %v", err)
+}
+
+// Get the channel
+channel := client.Channels.Get("{{RANDOM_CHANNEL_NAME}}")
+
+// Publish a message
+err = channel.Publish(context.Background(), "example", "message data")
+if err != nil {
+  log.Fatalf("Failed to publish message: %v", err)
+}
+
+// Attach the channel
+_ = channel.Attach(context.Background())
+
+// Get channel history until attach
+history, _ := channel.HistoryUntilAttach()
```

Comment:
Have we tested this explicitly in a separate example repo?
We don't have tests available for the same and only method was added as a part of protocol 2 work.
You can check method doc to test it explicitly.

Reply:
I have tested every code snippet that I've added. But I haven't got a separate example repo for testing. I did this with PHP, but it was too time consuming to do for every language.

Reply:
No issues, I assume you have tested `HistoryUntilAttach` that returns message history before attach only